src/pip/_internal/models/wheel.py

```diff
@@ -66,10 +66,31 @@ def support_index_min(self, tags):
         """
         return min(tags.index(tag) for tag in self.file_tags if tag in tags)
 
+    def find_most_preferred_tag(self, tags, tag_to_priority):
+        # type: (List[Tag], Dict[Tag, int]) -> int
+        """Return the priority of the most preferred tag that one of the wheel's file
+        tag combinations acheives in the given list of supported tags using the given
+        tag_to_priority mapping, where lower priorities are more-preferred.
+
+        This is used in place of support_index_min in some cases in order to avoid
+        an expensive linear scan of a large list of tags.
+
+        :param tags: the PEP 425 tags to check the wheel against.
+        :param tag_to_priority: a mapping from tag to priority of that tag, where
+            lower is more preferred.
+
+        :raises ValueError: If none of the wheel's file tags match one of
+            the supported tags.
+        """
+        return min(tag_to_priority[tag] for tag in self.file_tags if tag in tag_to_priority)
+
     def supported(self, tags):
-        # type: (List[Tag]) -> bool
+        # type: (Iterable[Tag]) -> bool
         """Return whether the wheel is compatible with one of the given tags.
 
         :param tags: the PEP 425 tags to check the wheel against.
         """
-        return not self.file_tags.isdisjoint(tags)
+        for itag in self.file_tags:
+            if itag in tags:
+                return True
+        return False
```

Comment:
```suggestion
        return not self.file_tags.isdisjoint(tags)
```

This should work, I think?

Reply:
Which is what the original function was. :)